dbfit-java/core/src/main/java/dbfit/api/AbstractDbEnvironment.java

```diff
@@ -50,6 +50,9 @@ protected void afterConnectionEstablished() throws SQLException {
 
     public void connect(String connectionString, Properties info) throws SQLException {
         registerDriver();
+        if (currentConnection != null) {
```

Comment:
No need of that `null` check here - `closeConnection()` already takes care about that.